lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -283,14 +287,149 @@ struct ConvolutionToConv2dPattern
+  }
+
+  mlir::DenseI64ArrayAttr
+  addIntegersToDenseArrayAttr(ConversionPatternRewriter &rewriter,
+                              mlir::DenseI64ArrayAttr denseArrayAttr,
+                              std::vector<uint64_t> additionalIntegers) const {
+    llvm::SmallVector<int64_t, 4> newDenseArray(denseArrayAttr.asArrayRef());
+    for (uint64_t integer : additionalIntegers) {
+      newDenseArray.push_back(integer);
+    }
+    return rewriter.getDenseI64ArrayAttr(newDenseArray);
+  }
+
+  mlir::DenseBoolArrayAttr
+  addBooleansToDenseArrayAttr(ConversionPatternRewriter &rewriter,
+                              mlir::DenseBoolArrayAttr denseArrayAttr,
+                              std::vector<bool> additionalIntegers) const {
+    llvm::SmallVector<bool, 4> newDenseArray(denseArrayAttr.asArrayRef());
+    for (bool boolean : additionalIntegers) {
+      newDenseArray.push_back(boolean);
+    }
+    return rewriter.getDenseBoolArrayAttr(newDenseArray);
+  }
 
-    if (failed(isConv2d(op))) {
+  LogicalResult matchAndRewrite1d(ttir::ConvolutionOp op, OpAdaptor adaptor,
+                                  ConversionPatternRewriter &rewriter) const {
+    // Not currently supporting spatial dims other than 2 for the 1D case.
+    if (op.getConvolutionLayout().getInputSpatialDimensions()[0] != 2) {
       return failure();
     }
 
+    auto outputType =
+        mlir::cast<RankedTensorType>(adaptor.getOutput().getType());
+    llvm::ArrayRef<int64_t> outputShape = outputType.getShape();
+    llvm::SmallVector<int64_t, 8> newOutputShape(outputShape.begin(),
+                                                 outputShape.end());
+    newOutputShape.push_back(1);
+    auto newOutput = rewriter.create<tensor::EmptyOp>(
+        op->getLoc(), newOutputShape, outputType.getElementType());
+    auto newOutputType = mlir::cast<RankedTensorType>(newOutput.getType());
+
+    auto inputType = mlir::cast<RankedTensorType>(adaptor.getInput().getType());
+    llvm::ArrayRef<int64_t> inputShape = inputType.getShape();
+    llvm::SmallVector<int64_t, 8> newInputShape(inputShape.begin(),
+                                                inputShape.end());
+    newInputShape.push_back(1);
+
+    auto weightType =
+        mlir::cast<RankedTensorType>(adaptor.getWeight().getType());
+    llvm::ArrayRef<int64_t> weightShape = weightType.getShape();
+    llvm::SmallVector<int64_t, 8> newWeightShape(weightShape.begin(),
+                                                 weightShape.end());
+    newWeightShape.push_back(1);
+
+    ttir::ReshapeOp reshapeInput =
+        createReshapeOp(op.getLoc(), adaptor.getInput(), newInputShape,
+                        op.getOperandConstraints(), rewriter);
+    ttir::ReshapeOp reshapeWeight =
+        createReshapeOp(op.getLoc(), adaptor.getWeight(), newWeightShape,
+                        op.getOperandConstraints(), rewriter);
+
+    mlir::DenseI64ArrayAttr newWindowsStridesAttr = addIntegersToDenseArrayAttr(
```

Comment:
`new` is not a great prefix, you use it for multiple ops here (e.g. reshape above also has `newOutputShape`), consider being more specific in naming.